pkg/datastreams/live_messages.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	kafkaConsumerIntegrationName = "kafka_consumer"
+)
+
+type kafkaConfig struct {
+	Cluster                 string `yaml:"cluster" json:"cluster"`
+	Topic                   string `yaml:"topic" json:"topic"`
+	Partition               int32  `yaml:"partition" json:"partition"`
+	StartOffset             int64  `yaml:"start_offset" json:"start_offset"`
+	NMessages               int32  `yaml:"n_messages" json:"n_messages"`
+	ValueFormat             string `yaml:"value_format" json:"value_format"`
+	ValueSchema             string `yaml:"value_schema" json:"value_schema"`
+	ValueUsesSchemaRegistry bool   `yaml:"value_uses_schema_registry" json:"value_uses_schema_registry"`
+	KeyFormat               string `yaml:"key_format" json:"key_format"`
+	KeySchema               string `yaml:"key_schema" json:"key_schema"`
+	KeyUsesSchemaRegistry   bool   `yaml:"key_uses_schema_registry" json:"key_uses_schema_registry"`
+}
+
+type liveMessagesConfig struct {
+	Kafka kafkaConfig `yaml:"kafka" json:"kafka"`
+	ID    string      `yaml:"id" json:"id"`
+}
+
+// Controller listens to remote configuration updates for the Data Streams live messages feature
+// and configures the kafka_consumer integration to fetch messages from Kafka.
+type Controller struct {
+	ac            autodiscovery.Component
+	configChanges chan integration.ConfigChanges
+	closeMutex    sync.RWMutex
+	closed        bool
+}
+
+// String returns the name of the provider.  All Config instances produced
+// by this provider will have this value in their Provider field.
+func (c *Controller) String() string {
+	return names.DataStreamsLiveMessages
+}
+
+// GetConfigErrors returns a map of errors that occurred on the last Collect
+// call, indexed by a description of the resource that generated the error.
+// The result is displayed in diagnostic tools such as `agent status`.
+func (c *Controller) GetConfigErrors() map[string]providers.ErrorMsgSet {
+	return nil
+}
+
+// ManageSubscriptionToRC subscribes to remote configuration updates if the agent is running the kafka_consumer integration.
+func (c *Controller) ManageSubscriptionToRC(subscribe func()) {
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+	for range ticker.C {
+		c.closeMutex.RLock()
+		if c.closed {
+			c.closeMutex.RUnlock()
+			return
+		}
+		c.closeMutex.RUnlock()
+		if isConnectedToKafka(c.ac) {
+			subscribe()
+			return
+		}
+	}
+}
+
+func isConnectedToKafka(ac autodiscovery.Component) bool {
+	for _, config := range ac.GetAllConfigs() {
+		if config.Name == kafkaConsumerIntegrationName {
+			return true
+		}
+	}
+	return false
+}
+
+// NewController creates a new Controller instance
+func NewController(ac autodiscovery.Component) *Controller {
+	return &Controller{
+		ac:            ac,
+		configChanges: make(chan integration.ConfigChanges, 10),
```

Comment:
Not a blocker, but I want to know if there's a particular reason why 10 was chosen?

Reply:
No reason tbh, something small enough not to use memory, but I don't want to block anything because a few config updates arrive at the same time.